### DISCUSSION TITLE
Android backup on the backgroud doesn't work

### DISCUSSION THREAD

**Discussion by podhorsky-ksj** [immich-app/immich#9205] [Q&A]
### The bug

Android backup on the background doesn't work, I have to open the app to start synchronization.
Allow background usage in settings is allowed. Nothing else like "no optimization" is not available.  

### The OS that Immich Server is running on

Arlinux

### Version of Immich Server

1.103

### Version of Immich Mobile App

1.103

### Platform with the issue

- [ ] Server
- [ ] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
Not applicable. Using https://aur.archlinux.org/packages/immich-server.
Backup is working, but not on the background, so it is not issue in server.
```

### Your .env content

```Shell
Not applicable.
```

### Reproduction steps

```bash
1.set backup on background in android app and backup only when wifi is available
2.close app
3.take photo of something
4. wait
5. nothing will happened, until I open the app. Then photo is uploaded.
```

### Additional information

Android is 14, clean lineage os for whyred (xiaomi redmi note 5 pro)

**Comment by mmomjian** [comment]
Are there any errors in the Immich logs? In general We have no control over this. https://immich.app/docs/FAQ/#why-does-foreground-backup-stop-when-i-navigate-away-from-the-app-shouldnt-it-transfer-the-job-to-background-backup

**Reply by techmattr** [reply]
I experience a similar or same issue on two different Samsung Galaxy phones. My S22 Plus and my wife's S23. The mobile app successfully uploads **new** photos that are taken if the app is in the background. But mainly the initial sync or a new album add do not upload unless I have the app open and I'm watching the backup progress screen. If I even switch to a different screen in the app, let the phone screen timeout or switch to a different app the upload stops. It immediately resumes uploading if I go back to the backup progress screen.

**Comment by podhorsky-ksj** [comment]
> Are there any errors in the Immich logs? In general We have no control over this. https://immich.app/docs/FAQ/#why-does-foreground-backup-stop-when-i-navigate-away-from-the-app-shouldnt-it-transfer-the-job-to-background-backup

There are a lot of errors in logs, but not from the time until the app was opened. It was 22:44-22:46

[Immich_log_2024-05-01T22_47_29.404099.log](https://github.com/immich-app/immich/files/15181260/Immich_log_2024-05-01T22_47_29.404099.log)

So that probably mean the app was suspended, when was not visible.

Can't you really do anything about it? Some apps are using wake lock - e.g termux and it works on background perfectly. It shows icon in upper panel, close to clock.

**Comment by masterwishx** [comment]
Have same issue in j710f android 8.1, disabled optimization of battery, like made for nextcloud and nextcloud sync is working but not immich...

**Comment by brianmiller** [comment]
Same issue here. I've disabled battery optimizations and disabled the ability for android to pause background activity.  The app only backups up when the app is open and the backup screen is up.

**Reply by mikelagase** [reply]
Same issue here on Samsung S21. The only thing I see in the server logs is this when my phone screen goes to sleep and it locks my phone but even then Immich doesn't log when a file is uploaded in the logs even with Debug logging. 

LOG [Api:EventRepository] Websocket Disconnect: aQO-qaz7P1CR7LulAAC5

Once I unlock my phone again it starts syncing again and I see the connect again

LOG [Api:EventRepository] Websocket Connect:    s5WqFHcfYTR-1mzdAAC7

This is definitely something up with Immich Android as the screen locking should not cause this to stop syncing especially when background sync is enabled.

**Comment by gomaaz** [comment]
Problem still persists with the newest app
App Version 1.109.2
Google Pixel 7
All Battery Optimizations off
App Grants set to be functional in background

**Comment by gomaaz** [comment]
update:

my experiments show a result:

- deselect all albums to be synced.
- go back to main screen of the app (timeline)
- go back to Sync - Choose desired albums
- close the app

After that background syncing starts.

Can someone reproduce?

**Reply by SimonTeixidor** [reply]
This worked for me, thanks.

**Reply by dasrohr** [reply]
Can confirm. This worked for me, too.

I also tried the usual other things the internet tells you to try beforehand. 
I am on a Samsung s22 Ultra (Android 14 OneUI 6.1)
No battery restictions, using Samsung's "Good Guardians" thingy, using "Keep open" feature from the recent apps menu and such.

So hard to tell if it was only the reconfiguring for the backup, or a combination of things. But it now says "background backup in progress" in the back overview, which it did not before.

Thanks for the hint.

**Reply by kimdre** [reply]
Thank you, also worked for me.
I wonder what causes that. 
It worked for me before when I set it up the first time and let the background service run over night, but after I changed some of the backup albums and settings it no longer worked until I reconfigured everything.

**Reply by kartikgola** [reply]
Thanks, but it only worked for a few mins before background sync stopped again

**Comment by bombela** [comment]
Same problem on the latest version (1.132.3 build.197).

Pixel 5a. Foreground backup off. Background backup on. Allow background usage on. Pause if unused off.

After opening Immich. Background backup will run for a while. Maybe a few days. And at some point, it will stop working.

For example today I noticed that the last upload was from a week ago even though I took pictures almost every day. Starting the Immich app awoken it, and now the background backup is working.

Google photo doesn't have this issue. I also have Syncthing uploading the photos in the background as well. (Yes I know, I am uploading every thing 3 times. Ultimately I want to rely on Immich only).

I don't have logs or actionable items to offer. Merely adding my observation to the list.

**Reply by calderonth** [reply]
I also can confirm this is what I am seeing on my Pixel 6a (latest Android/Immich app/server).

**Comment by bombela** [comment]
To add to it, I was on WiFi. Internet went down. Phone switched to cellular
temporarily. I took photos. When it was back on WiFi, Immich never resumed
backing up. Opening the application resumed the backup process right away.

**Comment by charettepa** [comment]
I am partially having the same issue

I have my battery settings for Immich set to "do not optimize" and i allow full unrestricted background use
even if i am not in the app, it DOES backup in the background

but if the screen turns off, it stops immediately

some people think google photos has "special" permission, as its Google
and that's why it works
but.... there is 100% has to be a way to get around this
I am currently migrating from Synology photos
it functions at all times even with screen off
so there has to be a way to achieve this, be it special app permission in the background or specific api calls, etc....

**Comment by bombela** [comment]
I wonder if this is because somehow immich doesn't get to see the new files
at all. On android 13, sync thing cannot see the files at all after the
initial scan. I haven't debugged far, but it literally do not see any new
files, it's like they do not exist. As if it is given an old view of the
filesystem. Adding new files manually via a shell did not change this view.
I could see them, see my new files. The file explorer on this flavor of
Android could seen my shell create files. But not synctning. Resetting the
state of sync thing and then going to the permission grant to access the
directory gave access to the files to syncthing. But any new files after
that were invisible.

Could it be some sort of android security somehow presenting a stale view
of the filesystem?

But after saying all that I am aware that Immich uses the photo API.

Not sure if helping but that's something.

**Comment by AngryJKirk** [comment]
Android 16, Pixel 9 Pro, still no changes. Everyday I just open the app to backup the data, otherwise it can be stale for weeks. 
Everything is on and every battery optimization available in Pixel is off.

**Reply by gamb1t9** [reply]
Pixel 7a but same

**Reply by maxiking445** [reply]
Same here. Nothing Phone 1&2

**Reply by KieranFletcher** [reply]
Pixel 9 Pro, still stuck

**Reply by Quemaqua** [reply]
Pixel 6 here, neither foreground or background automatic backups work for me. Have to manually start the process. Usual suspects turned on/off on phone (sleep, battery optimization, etc.).

**Reply by mptr8956** [reply]
Disabled beta timeline and enabled sync with old timeline. Waited to complete. Then switched back. That helped here to get it running again.

**Reply by oboeglen** [reply]
> Disabled beta timeline and enabled sync with old timeline. Waited to complete. Then switched back. That helped here to get it running again.

Confirmed, disabled beta and switch back works, crazy !

**Comment by bombela** [comment]
pixel5a. opening the app is enough to refresh and trigger the background
backup.

**Comment by henfri** [comment]
Hi, I Love Immich, but without a reliable Backup, I cannot use it.

Any fix in sight?

**Comment by maxiking445** [comment]
> Hi, I Love Immich, but without a reliable Backup, I cannot use it.
> 
> Any fix in sight?

With the new Beta it is way better for now. Beta is rolled out to everyone.

**Comment by oboeglen** [comment]
Same problem on Nothing Phone 3a, I need to start the app for backup process begin

**Comment by bombela** [comment]
The latest version of the android app; presumably after an update;
deactivated all backups. No warning. No nothing. That is really not cool.

I suspect that it used to backup the Camera folder without any specific
configuration. But now you have to explicitly add it to the list of folders
to backup? It's not really clear to me.

**Comment by bombela** [comment]
After selecting the camera folder for backup explicitly, it shows backup as
disabled. I have never disabled backups. The whole reason I am part of this
thread is because backup don't always work. But here it doesn't just
randomly not work. The application decided I didn't deserve backup one day
to the next.

**Reply by masterwishx** [reply]
> After selecting the camera folder for backup explicitly, it shows backup as disabled. I have never disabled backups. The whole reason I am part of this thread is because backup don't always work. But here it doesn't just randomly not work. The application decided I didn't deserve backup one day to the next. On Sat, Sep 20, 2025, 15:35 François-Xavier Bourlet ***@***.***> wrote:
> […](#)
> The latest version of the android app; presumably after an update; deactivated all backups. No warning. No nothing. That is really not cool. I suspect that it used to backup the Camera folder without any specific configuration. But now you have to explicitly add it to the list of folders to backup? It's not really clear to me. On Sat, Sep 20, 2025, 14:51 Olivier ***@***.***> wrote: > Same problem on Nothing Phone 3a, I need to start the app for backup > process begin > > — > Reply to this email directly, view it on GitHub > <[#9205 (comment)](https://github.com/immich-app/immich/discussions/9205#discussioncomment-14466094)>, > or unsubscribe > <https://github.com/notifications/unsubscribe-auth/AABUZDEC3UMZI3MKP74OBJ33TXD4PAVCNFSM6AAAAAB35NY6LGVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTINBWGYYDSNA> > . > You are receiving this because you commented.Message ID: > ***@***.***> >

I think new option was added that needs to be enabled in android app (Turn On Backup).
After that all working fine again on android 15 (A56 5G Samsung)

**Reply by oboeglen** [reply]
I’ve uninstalled and reinstalled the application on Android 15, enabled the backup process for my camera folder, configured battery usage to “unrestricted,” etc. Still, the backup only runs when I open the app. In my case, I regularly look at my photos, but it’s still problematic because as soon as the app is killed, nothing gets backed up anymore. Both my server and my app are on the latest version — when can we expect a truly functional backup that works in the background?

**Reply by masterwishx** [reply]
Maybe you have some app that kill immich?

**Reply by oboeglen** [reply]
Nope, I check all 
> Maybe you have some app that kill immich?

**Reply by masterwishx** [reply]
> Nope, I check all
> 
> > Maybe you have some app that kill immich?

I checked now again and has some issue until I changed to unrestricted battery for immich.

Then I closed all apps, took photos and video and they appeared in server. 

Also in android 15 there is adaptive mode for battery if app is using optimized mode it can be goes to sleep or deep sleep maybe it's related to what you select optimized etc, but can be check if immich is not in sleep list.

**Reply by oboeglen** [reply]
I've disabled everything and i'm in unrestricted battery for immich. Problem persist

> > Nope, I check all
> > 
> > > Maybe you have some app that kill immich?
> 
> I checked now again and has some issue until I changed to unrestricted battery for immich.
> 
> Then I closed all apps, took photos and video and they appeared in server. 
> 
> Also in android 15 there is adaptive mode for battery if app is using optimized mode it can be goes to sleep or deep sleep maybe it's related to what you select optimized etc, but can be check if immich is not in sleep list.

**Comment by bombela** [comment]
Android version 1.143.1 seems to backup much more consistently so far.

It also seems to periodically run a background scan, as indicated by an
Immich backup notification that comes up infrequently.

**Comment by tausen** [comment]
I had this issue, but it seems to be resolved after enabling notifications for immich in Android settings. I hadn't explicitly disabled them, so it was a bit of a surprise. Maybe it's the same for some of the other people here? 

Perhaps if the `immich::background_worker::notif` notification is not allowed to show, the app is not given any background time despite having all other requested permissions...

**Comment by NathanSweet** [comment]
It only uploads if the phone is unlocked and I'm watching the uploads. I can let a video upload get to 10%, lock the phone, unlock the phone, and see the upload restarts at 0%. Same happens switching apps, the queue resets. Android app version 2.5.6.

**It is completely unusable like this.**

I have Immich notifications enabled and syncing is still broken.

**Reply by passaca** [reply]
> It only uploads if the phone is unlocked and I'm watching the uploads. I can let a video upload get to 10%, lock the phone, unlock the phone, and see the upload restarts at 0%. Same happens switching apps, the queue resets. Android app version 2.5.6.
> 
> **It is completely unusable like this.**
> 
> I have Immich notifications enabled and syncing is still broken.

I have *exactly* the same problem. I tried everything that is mentioned and everything that comes to mind, but nothing worked.

When navigating away from the backup screen and coming back immediately, all progress numbers get this "loading circle" animation, meaning the backup was interrupted and is about to resume.

No progress is made while not directly looking at the backup screen.

I'm on a Oneplus phone but deactivated battery optimization for Immich. Other apps that need background processes (e.g. FairEmail) work flawlessly.

Any ideas?

PS: I checked the logs and they say (many times) "Backup was cancelled by the user" by "ForegroundUploadService".

Is this "normal"?

When would the background service start?

**Reply by NathanSweet** [reply]
It's broken. I left my phone on the Immich app every night until my thousands of photos were uploaded. Supposedly taking new pics to a backed up album (eg `Camera`) should trigger background sync, but it doesn't seem to be working either.

**Reply by passaca** [reply]
It must be working for most people, otherwise there would be more such threads here and the app rating would be way worse. 

But undoubtedly there are several such threads with "many" people reporting it, so the issue is definitely "real".

Now we need to determine what differentiates the people with issues from the people without issues.

People on Github are certainly very "tech-savvy", so I doubt that it is any simple config option.

**Reply by damomato** [reply]
Backup works fine on my phone but not my wife's (Samsung S23 FE) unless she keeps the app open. She keeps power saver on for whatever reason, I don't know I can't change behaviour, so I'm wondering if that's causing the issue for her.

I've turned off battery optimisation for Immich on her phone now and hopefully that will make a difference.

**Reply by Lixissimus** [reply]
I also have the exact same problem on Pixel 5. Im seeing only foreground sync mentions in the logs and it gets cancel by user once I leave the app or lock the screen. It's not usable like this unfortunately. I love everything else about immich, but this is very unfortunate.

**Reply by navehay** [reply]
I have the exact problem on s24. I have noticed that even in the middle of an upload of new asset found (that I put intently to test), in the time **before** I go out and immich is being killed by the OS, still immich is not showing as working in background as other 2 apps that stay all time in the background unless I forcelly stop them:
1. Netspeed app
2. Home front command (that gives me alarms on missiles that are on the way to me...)

These two apps has an icon on top near the time. Yet immich doesn't appear on top and doesn't appear as a background app.

Since These 2 apps works perfectly in the background unless I forcelly stop them, I suggest to use the same mechanism also for Immich to keep it working in background and save it from being killed by the aggressive Android OS.

<img width="1080" height="2340" alt="Screenshot_20260416_063955_One UI Home(1)" src="https://github.com/user-attachments/assets/9acba097-b96d-43e2-b3b6-af79a83eb77d" />

**Reply by navehay** [reply]
What I meant to say is why using foreground method and not background method as these other 2 apps the work perfectly for me on the background

**Comment by bombela** [comment]
Works for me on Android 14. Background usage allowed. (No battery
optimization option possible on Android 14 as far as I can tell)

**Comment by demario7** [comment]
I just noticed this issue as well and a Google search lead to this discussion as the first result. I have OnePlus 12 and sometimes it does reset my settings for different apps after an update (currently Android 16), however I made sure that it has all the permissions, automatic app management if unused is off and battery usage is unrestricted 'Allow background activity'.

Now I am just not sure if it worked this way the whole time because I could have sworn that background sync worked normally with my screen off before, it's possible I just didn't notice and it didn't sync prior either. Anyway, I recorded a short video to illustrate what's going on (it's 2x speed), although I am sure that's what everyone's issue is.

https://github.com/user-attachments/assets/e90235c0-ab9d-4bea-918a-de8adf2ac3ad

**Comment by axonasif** [comment]
Folks, I found a workaround that seems to work. 

You need to trigger some kind of FS operation within one of the selected albums. I could trigger it by deleting/moving/creating new photos within one of my albums and then this shows up:

<details><summary>Details</summary>
<p>

<img width="1440" height="3120" alt="2026-04" src="https://github.com/user-attachments/assets/4e4e3690-454e-49ed-a13b-5f46721f9fa4" />


</p>
</details> 

Note that it may take a while for this to show up as it depends on the fs watcher delay that you've configured in your app backup settings.

It'll become broken again if I open the app while it's doing the background upload but I can "fix" it by triggering some FS event again.


And just for context, whenever I move away from the app I see this warning popping up in the app logs:

```
warning  | ForegroundUploadService | Backup was cancelled by the user |
```

I'm running Immich v2.6.3 locally on macOS and the app version is also v2.6.3.

For now I'll use a dummy album from which I move a photo file in and out via something like MacroDroid to trigger this as a side effect.

Oh and since I'm on smasung, I had success with setting the app to "Never sleeping apps" list under battery settings rather than the native Android Settings app battery "Unrestricted" optimization option. https://dontkillmyapp.com is likely a helpful guide for other devices.

**Comment by damomato** [comment]
Wife upgraded her phone to Samsung s26+ and sync still not working for her unless the app is opened. She did the Samsung Sync transfer so if there was a setting causing this sync to work, that was likely transferred over.

When investigating how sync works on my own phone, I realised that while the image uploads fine without the app open, the image isn't added to the Camera album until the app is opened. Is that normal behavior? My wife is only backing up a single album (Camera) so not sure if this is contributing to the sync problems.

**Reply by alextran1502** [reply]
Do you have the notification turned on and the app has battery unrestricted?

**Reply by NathanSweet** [reply]
Yes. There's a huge number of people reporting this is broken. It's not a problem with battery saver killing it.

**Reply by damomato** [reply]
Ok so I went through the app settings and changed the battery to be unrestricted (I thought this was already set but I guess not).

Automatic backup is finally working on her phone 😃

I believe it was like this on her old phone, so I think this problem was mainly resolved from having a new phone and enabling that setting. The app should really have a reminder or option to enable it.

**Comment by navehay** [comment]
I have a feeling that the problem is because of aggressive RAM Management, and not because of aggressive Battery Optimization.

**Comment by ringe** [comment]
I just got my Immich 2.7.5 server up (container, local network) and am running a complete import from my Android Camera  in the latest app (from Google Play). Sync stops when I close the app.

Android 16, upd April 5th. Google Pixel 9 Pro.
Immich app 2.7.4

No battery saving, no Goggle deactivate, allow background work and unlimited data.

Charging the phone while while waiting on sync.

The app only perform backup while the screen is active, app is in focus.

**Reply by ringe** [reply]
I'm using the offical docker compose: https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml

**Reply by JoeyGH** [reply]
Same setup here, on the phone end I have Galaxy S24+ with Android 16 and uploads only work when app is in foreground and screen on. I have "Camera" folder synced and if I take a new photo, the notification "Immich checking for new..." appears, but it doesn't really do anything other than just show the notification. Then as soon as I open the app, the picture(s) get synced immediately.

**Reply by alextran1502** [reply]
For v2.7.5, background upload only runs for newly taken assets. In the future v3.0.0 release, it is now changed to always run for all assets

**Reply by fokkedenoord** [reply]
Title: Background backup never uploads new photos on Galaxy S24+ (Android 16, One UI 8.5) — foreground works fine
Environment:

Device: Samsung Galaxy S24+
OS: Android 16, One UI 8.5
Immich mobile app: 3.0.1 (build 3054)
Immich server: 3.0.1
Connection: Tailscale (MagicDNS hostname), server reachable and confirmed working

Bug description:
New photos taken on-device never get backed up in the background. They only upload once the Immich app is manually opened, at which point they sync immediately. Backup album selection is limited to Camera only (WhatsApp albums excluded). Background upload notification does not appear at all after taking a new photo and leaving the app; no upload activity is observed in the notification shade even after waiting 10–15 minutes with screen off.
Troubleshooting already performed (no change in behavior):

Battery: set to Unrestricted for Immich
Background data usage: allowed (both Wi-Fi and cellular)
Network access (Wi-Fi + mobile data): both allowed for Immich
Data Saver: confirmed off
Samsung Background usage limits: Immich added to "Never sleeping apps," confirmed not in Sleeping/Deep sleeping lists
Notification permission: enabled (was previously the source of an in-app warning, which disappeared after enabling)
"Manage app if unused" (app hibernation): disabled for Immich
Adaptive battery: checked
Device restarted after applying all settings above
Server connectivity confirmed not to be the cause: replaced MagicDNS hostname with the raw Tailscale IP (100.x.x.x) in server endpoint — no change
Confirmed Tailscale VPN remains "Connected" throughout the test window (background sync still fails despite active tunnel)
"Alarms and reminders" special access: Immich does not appear in this list (not applicable)
Cache cleared, logged out and back in — no change

Expected behavior: New photos in the selected backup album should upload automatically in the background, per the v3.0.0 release notes describing a "new periodic task scheduler" for Android background backup.
Actual behavior: No background upload activity occurs until the app is manually opened.
Additional notes: This appears to match other reports of the same issue on Galaxy S24/S24+ with Android 16 (e.g. discussion #9205), suggesting a device/OS-version-specific issue with how the background worker is scheduled or granted execution time on recent Samsung firmware, rather than a permissions/configuration issue on the user's end.

**Comment by gesin2** [comment]
Hello everyone. Thank you to the developers for their hard work and contribution to such a wonderful product. I would like to share my observations regarding the background upload functionality:

**Issue:** Background upload task times out with large video files.

**Android 16
v3.0.0-pr28452 16 May**

**1) Timeout issue with large files**
The background task (`Notification: "Searching for new files"`) does not have enough time to upload large video files within the timeout (19 minutes). In my case, the app successfully uploads small files (photos) until a queue builds up with 3 videos, each 4 GB or larger. After 19 minutes, the background task fails with a timeout error, and then **the task restarts from the very beginning**, losing all previous upload progress. As a result, the process gets stuck on these 3 videos and endlessly burns through traffic with no success.

```log
2026-05-20 09:18:36.132485 | info     | BackgroundWorkerBgService | Android background upload started hashTimeout: 180s, backupTimeout: 19m |
2026-05-20 09:17:31.649016 | info     | BackgroundWorkerBgService | Cleaning up background worker |
2026-05-20 09:17:31.645475 | info     | BackgroundWorkerBgService | Android background upload completed in 1147s |
2026-05-20 09:17:31.645004 | warning  | ForegroundUploadService | Backup was cancelled by the user |
2026-05-20 09:17:31.641437 | warning  | UploadRepository     | Upload asset[1000003222] was cancelled |
2026-05-20 09:17:31.597961 | warning  | BackgroundWorkerBgService | Android background upload timed out after 19m, cancelling backup |
2026-05-20 08:58:31.724191 | info     | ForegroundUploadService | Network capabilities: [NetworkCapability.wifi, NetworkCapability.vpn, NetworkCapability.unmetered], hasWifi/isUnmetered: true |
2026-05-20 08:58:23.817519 | info     | BackgroundWorkerBgService | Android background upload started hashTimeout: 180s, backupTimeout: 19m |
```

**2) Ambiguous notification text during background upload**
During background upload, the notification in the drawer shows the ambiguous text: `Searching for new files`. This gives the average user the impression that the process is stuck and nothing is happening. It would be much better if the notification displayed the names of the files being uploaded and the upload progress percentage for each file.

**Comment by bigheadsmith** [comment]
Anyone who is seeing problems with immich app uploading in the background, do you also have any other apps set to upload the same media in the background?

I disabled Google photos and OneDrive photo backups and immich seems to be working fine now

**Comment by lewpbrown** [comment]
Make sure notifications are enabled for immich in your android settings - mine was toggled off, and when I toggled it on background sync started working as expected.
